### PR TITLE
Fix migrate-field-database-type-test flake

### PR DESCRIPTION
**Pull request by calherries**
Fixes a flake on the `migrate-field-database-type-test`, which is flaking on the be-tests-mysql-5-7-ee and be-tests-mysql-latest-ee workflows.

The flake was introduced [here](https://github.com/metabase/metabase/commit/48dcfc92d465e6f36b8e0f6ab37b1cb79aa32115).

This PR switches the use of [`metabase.db.schema-migrations-test.impl/run-migrations-in-range!`](https://github.com/metabase/metabase/blob/b912a8bdffe8a2fea7e5f67851dff500c9693166/test/metabase/db/schema_migrations_test/impl.clj#L159) with [`metabase.db.setup/migrate!`](https://github.com/metabase/metabase/blob/b912a8bdffe8a2fea7e5f67851dff500c9693166/src/metabase/db/setup.clj#L52), which is used in the [rollback test](https://github.com/metabase/metabase/blob/b912a8bdffe8a2fea7e5f67851dff500c9693166/test/metabase/db/schema_migrations_test.clj#L53) and seems to fix the issue. I'm not entirely sure why though, because I couldn't reproduce this locally.

### CODE REVIEW

**Comment by qnkhuat** [comment]
Example of a failing build: https://github.com/metabase/metabase/actions/runs/4507886907/jobs/7936085662?pr=29511

File: test/metabase/db/schema_migrations_test.clj [hunk 1]
```diff
@@ -966,8 +966,11 @@
 
 (deftest migrate-field-database-type-test
   (testing "Migration v47.00-003: set base-type to type/JSON for JSON database-types for postgres and mysql"
```
Comment (qnkhuat) [inline test/metabase/db/schema_migrations_test.clj:968]:
```suggestion
  (testing "Migration v47.00-001: set base-type to type/JSON for JSON database-types for postgres and mysql"
```

File: test/metabase/db/schema_migrations_test.clj [hunk 2]
```diff
@@ -965,9 +965,12 @@
...
 
 (deftest migrate-field-database-type-test
-  (testing "Migration v47.00-003: set base-type to type/JSON for JSON database-types for postgres and mysql"
-    (impl/test-migrations ["v47.00-001"] [migrate!]
-      (let [[pg-db-id
+  (testing "Migration v47.00-001: set base-type to type/JSON for JSON database-types for postgres and mysql"
+    (impl/test-migrations ["v47.00-001"] [_]
+      (let [{:keys [db-type ^javax.sql.DataSource
+                    data-source]} mdb.connection/*application-db*
+            migrate! (partial db.setup/migrate! db-type data-source)
```
Comment (qnkhuat) [inline test/metabase/db/schema_migrations_test.clj:972]: I'd have a comment here to why we use this

**Review by qnkhuat — APPROVED** [review]
what's a strange bug.

i too can't reproduce this locally but I think the change is safe.

File: test/metabase/db/schema_migrations_test.clj (hunk 2 above)
Comment (calherries) [reply test/metabase/db/schema_migrations_test.clj:974]: Done https://github.com/metabase/metabase/commit/3c14be2992a1c42fddf251aac26b53d87af329b5

**Pull request merged**